src/app/system/codes/view-code/view-code.component.ts

```diff
@@ -0,0 +1,215 @@
+      'name': [{ value: codeValue ? codeValue.name : '', disabled: true }, Validators.required],
+      'description': [{ value: codeValue ? codeValue.description : '', disabled: true }],
+      'position': [{ value: codeValue ? codeValue.position : 0, disabled: true }, Validators.required],
+      'active': [{ value: codeValue ? codeValue.active : false, disabled: true }]
+    });
+  }
+
+  /**
+   * Deletes the particular code value.
+   * @param {number} index Index of the row.
+   */
+  deleteCodeValue(index: number) {
+    const codeValueId = this.codeValuesData[index].id;
+    this.systemService.deleteCodeValue(this.codeData.id, codeValueId)
+      .subscribe((response: any) => { });
+    this.codeValuesData.splice(index, 1);
+    this.codevalues.removeAt(index);
+    this.codeValueRowStatus.splice(index, 1);
+  }
+
+  /**
+   * Removes/Deletes the particular code value which has not been intialised.
+   * @param {number} index Index of the row.
+   */
+  removeNewCodeValue(index: number) {
+    this.codevalues.removeAt(index);
+    this.codeValueRowStatus.splice(index, 1);
+  }
+
+  /**
+   * Updates the particular code value.
+   * @param {number} index Index of the row.
+   */
+  updateCodeValue(index: number) {
+    const updatedCodeValue: { name: string, description: string, position: number, isActive: boolean } = {
+      name: this.codevalues.at(index).get('name').value,
+      description: this.codevalues.at(index).get('description').value,
+      position: this.codevalues.at(index).get('position').value,
+      isActive: this.codevalues.at(index).get('active').value
+    };
+    this.systemService.updateCodeValue(this.codeData.id, this.codeValuesData[index].id, updatedCodeValue)
+      .subscribe((response: any) => {
+        this.codevalues.at(index).disable();
+        this.codeValueRowStatus[index] = 'disabled';
+        this.codevalues.at(index).markAsPristine();
+      });
+  }
+
+  /**
+   * Deletes the code.
+   */
+  delete() {
+    const deleteCodeDialogRef = this.dialog.open(DeleteDialogComponent, {
+      data: { deleteContext: `code ${this.codeData.name}` }
+    });
+    deleteCodeDialogRef.afterClosed().subscribe((response: any) => {
+      if (response.delete) {
+        this.systemService.deleteCode(this.codeData.id)
+        .subscribe(() => {
+          this.router.navigate(['../../'], { relativeTo: this.route });
+        });
+      }
+    });
+  }
+
+  /**
+   * Disables the particular row.
+   * @param {number} index Index of the row.
+   */
+  disableRow(index: number) {
+    this.codevalues.at(index).disable();
+    this.codeValueRowStatus[index] = 'disabled';
+    this.codevalues.at(index).markAsPristine();
+  }
+
+  /**
+   * Adds the given code value.
+   * @param {number} index Index of the row.
+   */
+  addCodeValue(index: number) {
+    const newCodeValue: { name: string, description: string, position: string, isActive: boolean } = {
+      name: this.codevalues.at(index).get('name').value,
```

Comment:
Try to access this object from the array directly instead of constructing it here again.

Reply:
Refer to the comment above.

Reply:
`codevalues`